ast/src/builder.rs

```diff
@@ -503,6 +503,15 @@ pub fn combine_imports(nodes: Vec<NodeData>) -> Vec<NodeData> {
     if nodes.is_empty() {
         return Vec::new();
     }
+
+    let file_path = &nodes[0].file;
+    let file_name = file_path.split('/').last().unwrap_or(file_path);
+
+    let import_name = if nodes.len() == 1 {
+        nodes[0].name.clone()
+    } else {
+        format!("imports-{}-{}", file_name, nodes.len())
```

Comment:
What if two files have the same name in different directories? Just use `create_synthetic_key_from_ref` or some other unique key here